### ISSUE TITLE
Speedup RBin loading times

### ISSUE THREAD

**Issue by radare** [radare/radare2#8069]
RBin is an important piece of r2, and for some big binaries it takes so much time to load, the reasons are various, but the main reason of this is because symbols are walked several times.

* rbin backend generates list of symbols
* rbin converts this list into an rlist, performs demangling, filter dupes and such
* rcorebin creates the flags accordingly

Sometimes this process is done twice or more times.

**Comment by radare** [comment]
Proper way to speedup this would be to use Sdb from RBin, storing all symbols in a hashtable.

* rlist is implicit, so no more list -> rlist conversion needed
* detect 2 symbols with the same name at startup time
* avoid having to cache symbols in ht for later resolving imports (the has_lit thing)

**Comment by ret2libc** [comment]
By "sdb" you mean just an hashtable, right?

**Comment by radare** [comment]
well the idea of using sdb, is to have by-de-facto disk hashtables and quick project savings. ht doesnt supports this

**Comment by XVilka** [comment]
For porting to the RVector too

**Comment by ret2libc** [comment]
Just for clarity... better not use Sdb, but just an HT where needed. Variables in RAnal is a good proof of why sdb is not good when used directly.. it shall just be used to serialize/deserialize stuff.

**Comment by thestr4ng3r** [comment]
Yeah, the benefit that you have "quick projects" is killed by the fact that you would have to still write the serialization/deserialization code, but at runtime, making everything bloated and slow. It really only makes sense when the string kv fits the data structure by design.

**Comment by radare** [comment]
well. the reason for using sdb in projects was because tloading all the data in a disk hashtable takes 0 time. there's no other solution for this outside sdb right now
